src/worker.js

```diff
@@ -174,7 +174,13 @@ this.onmessage = function(e) {
 #if STACK_OVERFLOW_CHECK
       Module['writeStackCookie']();
 #endif
-
+#if EMBIND
+      // Embind must initialize itself on all threads, as it generates support JS.
+      // We only do this once per worker since they get reused
+      if (Object.keys(registeredTypes).length === 0) {
+        Module['___embind_register_native_and_builtin_types']();
```

Comment:
But are there not native C++ data structures that need to be re-created each time the worker is re-used?     Or does this function only have JS side effects?

Reply:
It's just a bunch of calls to `_embind_register_xxx` passing in various sizeof() and limits etc so there shouldn't be any native side-effects. It's just done from native since it's compiler dependent I guess.